test/utils/pickers-utils.tsx

```diff
@@ -74,6 +80,123 @@ export function createPickerRenderer({
   };
 }
 
-export function openMobilePicker() {
-  fireEvent.click(screen.getByRole('textbox'));
-}
+type OpenPickerParams =
+  | {
+      type: 'date' | 'date-time' | 'time';
+      variant: 'mobile' | 'desktop';
+    }
+  | {
+      type: 'date-range';
+      variant: 'mobile' | 'desktop';
+      initialFocus: 'start' | 'end';
+    };
+
+export const openPicker = (params: OpenPickerParams) => {
+  switch (params.variant) {
+    case 'desktop': {
+      switch (params.type) {
+        case 'date':
+        case 'date-time': {
+          return userEvent.mousePress(screen.getByLabelText(/choose date/i));
+        }
+
+        case 'time': {
+          return userEvent.mousePress(screen.getByLabelText(/choose time/i));
+        }
+
+        case 'date-range': {
+          return fireEvent.focus(
+            screen.getAllByRole('textbox')[params.initialFocus === 'start' ? 0 : 1],
+          );
+        }
+
+        default: {
+          throw Error('Invalid params');
+        }
+      }
+    }
+
+    case 'mobile': {
+      switch (params.type) {
+        case 'date':
+        case 'date-time':
+        case 'time': {
+          return userEvent.mousePress(screen.getByRole('textbox'));
+        }
+
+        case 'date-range': {
+          return userEvent.mousePress(
```

Comment:
Why the difference between mobile and desktop? Could `userEvent.focus` be used here?